index.js

```diff
@@ -102,15 +102,21 @@ var Database = function (options) {
 			});
 	};
 
-	var pool = mysql.createPool({
-		host: options.hostname,
-		user: options.username,
-		password: options.password,
-		database: options.database,
+	var parsedDsn = {};
+	if (options.dsn) {
+		parsedDsn = require("mysql/lib/ConnectionConfig").parseUrl(options.dsn);
+	}
+
+	var poolConfig = {
+		host: options.hostname || parsedDsn.host,
```

Comment:
Nah. Who cares. It's either/or in reality - if both are provided - it's the user's problem, not mine.